src/components/UserEventLocation/UserEventLocation.tsx

```diff
@@ -1,39 +1,59 @@
+import { useEffect } from 'react';
 import { ReactSVG } from 'react-svg';
 import styles from './UserEventLocation.module.scss';
 import { LectureEvent } from '../../domain/lectureEvent';
 import busIcon from '../../assets/icons/busIcon.svg';
 import carIcon from '../../assets/icons/carIcon.svg';
 import { Faculty, generateGoogleMapsLink } from '../../helpers/googleMaps';
+import { useAuthStore } from '../../stores/auth/auth.store';
 
 type Props = {
   lectureEvent: LectureEvent;
   faculty: Faculty;
 };
 
 const UserEventLocation = ({ lectureEvent, faculty }: Props) => {
+  const { currentUser, getCurrentUser } = useAuthStore();
+
+  useEffect(() => {
+    getCurrentUser();
+  }, []);
+
   return (
     <div className={styles.locationClassWithIconsConteiner}>
       <p>{lectureEvent?.location}</p>
-      <ReactSVG
-        src={busIcon}
-        className={styles.iconsForGoogleMaps}
-        onClick={() =>
-          window.open(
-            generateGoogleMapsLink(faculty, lectureEvent.startDateTime, 'bus'),
-            '_blank',
-          )
-        }
-      />
-      <ReactSVG
-        src={carIcon}
-        className={styles.iconsForGoogleMaps}
-        onClick={() =>
-          window.open(
-            generateGoogleMapsLink(faculty, lectureEvent.startDateTime, 'car'),
-            '_blank',
-          )
-        }
-      />
+      <div className={styles.iconsContainer}>
+        <ReactSVG
+          src={busIcon}
+          className={styles.icon}
+          onClick={() => {
+            window.open(
+              generateGoogleMapsLink(
```

Comment:
change the name of this function to better reflect what it does. This function can generate not only a link for google maps but also for waze. "getNavigationLink" would work or you can think of something else